src/include/migraphx/op/prefix_scan_op.hpp

```diff
@@ -0,0 +1,104 @@
+#ifndef MIGRAPHX_GUARD_OPERATORS_SCAN_OP_HPP
+#define MIGRAPHX_GUARD_OPERATORS_SCAN_OP_HPP
+
+#include <migraphx/op/name.hpp>
+#include <migraphx/check_shapes.hpp>
+#include <migraphx/argument.hpp>
+#include <migraphx/shape_for_each.hpp>
+#include <migraphx/par_for.hpp>
+#include <migraphx/config.hpp>
+#include <migraphx/value.hpp>
+#include <migraphx/op/normalize_attribute.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+template <class Derived>
+struct prefix_scan_op : op_name<Derived>
+{
+    int64_t axis;
```

Comment:
I think the axis had better be an input, not an axis. Then axis input for this operator does not have to be constant. 
The reason we put axis as an attribute for other operators (like gather, concat) is that different axis values generates different output shapes, and migraphx cannot support that. But this operator is different. Its output shape is always the same, so the axis can be dynamic.